IPs/RSKIP176.md

```diff
@@ -0,0 +1,145 @@
+
+## Motivation
+
+Allowing users to transfer BTC to RSK immediately, without waiting for enforced peg-in confirmation blocks. This also generates a market between user who wants to peg-in faster and those whose who provide the liquidity to advance the payment and accept the risk of the reversal of the Bitcoin blockchain.
+
+## Specification
+
+The fast-bridge system connects users, Liquidity Providers (LP) and a certain Liquidity Bridge Contract (LBC).
+This RSKIP details the specification of consensus changes to achieve the main goal of the use case without limiting the actual implementation of the fast-bridge system.
+
+#### Bridge
+
+The bridge needs to expose a new method called **registerFastBridgeBtcTransaction**().
+
+#### registerFastBridgeBtcTransaction()
+
+**ABI Signature**
+
+        function registerFastBridgeBtcTransaction(bytes btcTxSerialized, uint256 height, bytes pmtSerialized, bytes32 derivationArgumentsHash, string userRefundBtcAddress, string LiquidityBridgeContractAddress, string LiquidityProviderBtcAddress, bool shouldTransferToContract) returns int executionStatus
+
+
+**Derivation Arguments**
+
+The derivation arguments are values used to close the agreement between the Liquidity Provider and the user. Some of these parameters are the RSK address where funds will be received, the BTC refund address, the Liquidity Bridge Contract address, the value to transfer, the amount of BTC blocks that the Liquidity Provider should wait before advancing the funds, the Liquidity Provider BTC refund address, the derived Federation address, and some other values that are domain of the Liquidity Provider.
+
+
+**Parameters**:
+- *bytes* btcTxSerialized: Serialized Bitcoin transaction following Bitcoin serialization
+- *uint256* height: BTC block number where the transaction is present
+- *bytes* pmtSerialized: Serialized partial merkle tree (following bitcoinj structure) that proves that the transaction belongs to the indicated block
+- *bytes32* derivationArgumentsHash: A Keccak hash created from the derivation arguments
+- *bytes* userRefundBtcAddress: Binary encoding representing the user BTC refund address
+- *string* LiquidityBridgeContractAddress: String representing RSK Liquidity Bridge Contract address
+- *bytes* LiquidityProviderBtcAddress: Binary encoding representing  the Liquidity Provider BTC refund address
+- *bool* shouldTransferToContract: Status provided by the caller contract to know if should transfer value to the contract or not
+
+
+**Flow**:
+
+**Actions performed by the Bridge:**
+
+- Bridge receives a call from a contract (an alleged Liquidity Bridge Contract) and validates that the sender of that call is effectively the LBC address, that is received as parameter
```

Comment:
an "alleged" LBC? meaning of alleged is unclear

Reply:
Should be "(which is supposed to be a LBC)"